src/applications/ivc-champva/10-10D/config/form.js

```diff
@@ -1264,6 +1248,120 @@ const formConfig = {
+        },
+        // Applicant remarried after sponsor died
+        page18f4: {
+          path: 'applicant-information/:index/remarried-dates',
+          arrayPath: 'applicants',
+          showPagePerItem: true,
+          title: item => `${applicantWording(item)} marriage dates`,
+          depends: (formData, index) => {
+            if (index === undefined) return true;
+            return (
+              get(
+                'applicantRelationshipToSponsor.relationshipToVeteran',
+                formData?.applicants?.[index],
+              ) === 'spouse' &&
+              get(
+                'applicantSponsorMarriageDetails.relationshipToVeteran',
+                formData?.applicants?.[index],
+              ) === 'marriedTillDeathRemarriedAfter55' &&
+              get('remarriageIsViable', formData?.applicants?.[index])
+            );
+          },
+          uiSchema: marriageDatesSchema.uiSchema,
+          schema: marriageDatesSchema.remarriageSchema,
+        },
+        // Applicant remarried after sponsor died but separated from 2nd spouse
+        page18f5: {
+          path: 'applicant-information/:index/remarried-separated-dates',
+          arrayPath: 'applicants',
+          showPagePerItem: true,
+          title: item => `${applicantWording(item)} marriage dates`,
+          depends: (formData, index) => {
+            if (index === undefined) return true;
+            return (
+              get(
+                'applicantRelationshipToSponsor.relationshipToVeteran',
+                formData?.applicants?.[index],
+              ) === 'spouse' &&
+              get(
+                'applicantSponsorMarriageDetails.relationshipToVeteran',
+                formData?.applicants?.[index],
+              ) === 'marriedTillDeathRemarriedAfter55' &&
+              !get('remarriageIsViable', formData?.applicants?.[index])
+            );
+          },
+          uiSchema: marriageDatesSchema.uiSchema,
+          schema: marriageDatesSchema.remarriageSeparatedSchema,
+        },
+        // Applicant separated from sponsor before sponsor's death
+        page18f6: {
+          path: 'applicant-information/:index/married-separated-dates',
+          arrayPath: 'applicants',
+          showPagePerItem: true,
+          title: item => `${applicantWording(item)} marriage dates`,
+          depends: (formData, index) => {
+            if (index === undefined) return true;
+            return (
+              get(
+                'applicantRelationshipToSponsor.relationshipToVeteran',
+                formData?.applicants?.[index],
+              ) === 'spouse' &&
+              get(
+                'applicantSponsorMarriageDetails.relationshipToVeteran',
+                formData?.applicants?.[index],
+              ) === 'marriageDissolved'
+            );
+          },
+          uiSchema: marriageDatesSchema.uiSchema,
+          schema: marriageDatesSchema.separatedSchema,
+        },
```

Comment:
Added multiple pages that all use the same uiSchema that contains 4 date fields. Each of these pages uses a different schema though, which allows us to show or hide the appropriate date fields. 

Had to do this because `hideIf` relies on a widget supporting the `hidden` property, which the `currentOrPastDateUI` does not support, so we couldn't just use conditional functions to show/hide date fields within a single page.